## protocols

This directory contains laboratory protocols and SOPs (or links to them) for sample and metadata collection, sample tracking, amplicon sequencing, shotgun sequencing, and metabolomics.

### Overview

The first phase of the EMP (2010-2017) explicitly used the DNA extraction and amplicon sequencing protocols, but not necessarily the other protocols, described here. The 16S rRNA amplicon protocol in particular is described in detail by Caporaso et al. ([*PNAS*, 2011](http://doi.org/10.1073/pnas.1000080107)). The full methods used in the meta-analysis of this first phase (Release 1) are described by Thompson et al. ([*Nature*, 2017](http://doi.org/10.1038/nature24621)).

The second phase of the EMP (2016-present), a multi-omics effort informally called "EMP500", uses all of the protocols and SOPs presented here. EMP500 involves the metagenomic sequencing and metabolomic profiling, in addition to amplicon sequencing, of ~500 freshly collected environmental samples from diverse sites on our planet. This multi-omics dataset will be used to address multiple technical and ecological questions in microbial ecology. A biobank of frozen aliquots of samples are being maintained at UCSD and PNNL for future methods testing and analysis.

### Sample collection and aliquot tracking

Maintaining multiple (~10) individual frozen aliquots of each raw sample is suggested. This minimizes freeze-thaw cycles and permits multiple future downstream processing protocols. 

To track the provenance of these aliquots, we employ the following QR barcoding scheme:

* Tubes should be 2-mL screw-cap bead beater tubes to enable direct use in DNA extraction.
* Labels should be affixed to aliquot tubes before shipping. 
* QR codes have the format `doe.99.s003.a05`, where "doe" is the PI name, "99" is the study ID, "s003" is the sample number, and "a05" is the aliquot number. 
* QR codes (version 2, 25x25) are printed on Cryogenic Direct Thermal Labels, 1.125" x 0.75" rectangluar labels and 0.437" circular cap labels (GA International, part no. DFP-70) using a Zebra model GK420d printer and ZebraDesigner Pro software for Windows.
* Before aliquots are put away, QR codes are scanned into a sample inventory spreadsheet using a QR scanner.

### Metadata collection, curation, and processing

Metadata should comply with MIMS and Qiita standards. Please refer to the EMP500 [metadata guide](http://press.igsb.anl.gov/earthmicrobiome/protocols-and-standards/metadata-guide/).

### Amplicon sequencing

The standard EMP protocols for DNA extraction and 16S, 18S, and ITS amplicon sequencing are at [earthmicrobiome.org](http://www.earthmicrobiome.org/protocols-and-standards/), which has the following sections:

* [DNA Extraction Protocol](http://www.earthmicrobiome.org/protocols-and-standards/dna-extraction-protocol/)
* [Primer Ordering and Resuspension](http://www.earthmicrobiome.org/protocols-and-standards/primer-ordering-and-resuspension/)
* [16S Illumina Amplicon Protocol](http://www.earthmicrobiome.org/protocols-and-standards/16s/)
* [18S Illumina Amplicon Protocol](http://www.earthmicrobiome.org/protocols-and-standards/18s/)
* [ITS Illumina Amplicon Protocol](http://www.earthmicrobiome.org/protocols-and-standards/its/)

### Shotgun sequencing

(manuscript in prep)

### Metabolomics

(manuscript in prep)
